base_layer/wallet_ffi/src/lib.rs

```diff
@@ -232,34 +492,681 @@ pub unsafe extern "C" fn contact_get_alias(contact: *mut TariContact) -> *mut c_
+#[no_mangle]
+pub unsafe extern "C" fn pending_inbound_transaction_get_source_public_key(
+    transaction: *mut TariPendingInboundTransaction,
+) -> *mut TariPublicKey {
+    if transaction.is_null() {
+        return ptr::null_mut();
+    }
+    let m = (*transaction).source_public_key.clone();
+    Box::into_raw(Box::new(m))
+}
+
+/// Gets the amount of a TariPendingInboundTransaction
+///
+/// ## Arguments
+/// `transaction` - The pointer to a TariPendingInboundTransaction
+///
+/// ## Returns
+/// `c_ulonglong` - Returns the amount, note that it will be zero if transaction is null
+#[no_mangle]
+pub unsafe extern "C" fn pending_inbound_transaction_get_amount(
+    transaction: *mut TariPendingInboundTransaction,
+) -> c_ulonglong {
+    if transaction.is_null() {
+        return 0;
+    }
+    c_ulonglong::from((*transaction).amount)
+}
+
+/// Gets the timestamp of a TariPendingInboundTransaction
+///
+/// ## Arguments
+/// `transaction` - The pointer to a TariPendingInboundTransaction
+///
+/// ## Returns
+/// `c_ulonglong` - Returns the timestamp, note that it will be zero if transaction is null
+#[no_mangle]
+pub unsafe extern "C" fn pending_inbound_transaction_get_timestamp(
+    transaction: *mut TariPendingInboundTransaction,
+) -> c_longlong {
+    if transaction.is_null() {
+        return 0;
+    }
+    (*transaction).timestamp.timestamp() as c_longlong
+}
+
+/// Gets the message of a TariPendingInboundTransaction
+///
+/// ## Arguments
+/// `transaction` - The pointer to a TariPendingInboundTransaction
+///
+/// ## Returns
+/// `*const c_char` - Returns the pointer to the char array, note that it will return a pointer
+/// to an empty char array if transaction is null
+#[no_mangle]
+pub unsafe extern "C" fn pending_inbound_transaction_get_message(
+    transaction: *mut TariPendingInboundTransaction,
+) -> *const c_char {
+    let message = (*transaction).message.clone();
+    let mut result = CString::new("").unwrap();
+    if transaction.is_null() {
+        return result.into_raw();
+    }
+
+    result = CString::new(message).unwrap();
+    result.into_raw()
+}
+
+/// Destroys a TariPendingInboundTransaction
+///
+/// ## Arguments
+/// `transaction` - The pointer to a TariPendingInboundTransaction
+///
+/// ## Returns
+/// `()` - Does not return a value, equivalent to void in C
+#[no_mangle]
+pub unsafe extern "C" fn pending_inbound_transaction_destroy(transaction: *mut TariPendingInboundTransaction) {
+    if !transaction.is_null() {
+        Box::from_raw(transaction);
+    }
+}
 /// -------------------------------------------------------------------------------------------- ///
 
 /// ----------------------------------- CommsConfig ---------------------------------------------///
 
+/// Creates a TariCommsConfig
+///
+/// ## Arguments
+/// `address` - The address char array pointer
+/// `database_name` - The database name char array pointer
+/// `database_path` - The database path char array pointer
```

Comment:
add a mention that a db file will be created at this path to indicate to the API user that it should be a path to a location they have write access to.

Reply:
Updated in next commit